### DISCUSSION TITLE
Question: An extra blank line at the end of the code block

### DISCUSSION THREAD

**Discussion by ngmy** [thephpleague/commonmark#685] [Q&A] [answered]
I use `league/commonmark` version 1.6.4 and `league/html-to-markdown` version 5.0.0.

````
<?php

require __DIR__ . '/vendor/autoload.php';

use League\CommonMark\GithubFlavoredMarkdownConverter;
use League\HTMLToMarkdown\HtmlConverter;

$markdown = <<<MARKDOWN
```
hogehoge
```
MARKDOWN;

$markdownConverter = new GithubFlavoredMarkdownConverter([
    'html_input' => 'strip',
    'allow_unsafe_links' => false,
]);
$html = $markdownConverter->convertToHtml($markdown);

echo $html;

$htmlConverter = new HtmlConverter();
$markdown = $htmlConverter->convert($html);

echo $markdown;
````

The html generated during the process is as follows:

```
<pre><code>hogehoge
</code></pre>
```

The markdown generated at the end is as follows:
````
```
hogehoge

```
````

I have an extra blank line at the end of a code block.
Is there any way to prevent the extra blank line from being created?

**Comment by colinodell** [comment, marked as answer]
Good question!  The [CommonMark spec](https://spec.commonmark.org/0.30/#fenced-code-blocks) says:

> The content of a code fence is treated as literal text

Because `hogehoge` is followed by a newline character before the final fence, that newline gets included.

There is no built-in way to remove that trailing newline character, but you could implement a [custom event listener](https://commonmark.thephpleague.com/1.6/customization/event-dispatcher/#example) that listens for the `DocumentParsedEvent`, [walks the AST](https://commonmark.thephpleague.com/1.6/customization/abstract-syntax-tree/#iteration--walking-the-tree) looking for `FencedCode` notes, and then perform an `rtrim()`  on the contents.  Unfortunately, 1.x doesn't provide functionality to modify those contents, but 2.0 will!  For 1.x you'd instead need to implement a custom renderer for `FencedCode` objects and strip the trailing newline that way.

**Reply by ngmy** [reply]
Thank you for your answer.
I think I understand now.
